utils/lexicographical_compare.hh

```diff
@@ -0,0 +1,162 @@
+// Copyright 2023-present ScyllaDB
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+#pragma once
+
+#include <compare>
+#include <cstdint>
+#include <iterator>
```

Comment:
so we can at least remove `#include <iterator>` and probably `#include <compare>` from `types/types.hh`  ?

Reply:
#13327